pkg/controller/cronjob/utils.go

```diff
@@ -200,10 +198,9 @@ func getJobFromTemplate(sj *batch.CronJob, scheduledTime time.Time) (*batch.Job,
 	return job, nil
 }
 
-func getTimeHash(scheduledTime time.Time) uint32 {
-	timeHasher := adler32.New()
-	hashutil.DeepHashObject(timeHasher, scheduledTime)
-	return timeHasher.Sum32()
+// Return Unix Epoch Time
+func getTimeHash(scheduledTime time.Time) int64 {
+	return scheduledTime.Unix()
```

Comment:
The granularity of scheduled jobs is one second (because it's a cron like schedule).
The change is about replacing PODs suffix hash with the epoch time. So if you have many jobs, their prefixes will differ. I am not changing the prefixes.
Instead of:
```
etcd-backup-4175643772-g57gv
etcd-backup-4209853567-nlipe
```
you will have:
```
etcd-backup-1479081600-g57gv
etcd-backup-1479081900-nlipe
```